### PR TITLE
EVG-14708: set up ECS client interface

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-14708

This creates a new package with the ECS client interface and updates the tests to handle multiple packages. The client only deals with interacting with the ECS API. I thought it might be nice to have a manager to do common ECS operations, but it seems like the management would depend on the data model of the application. Since pail doesn't deal with application data models, I think the management layer should be written in Evergreen/Cedar, wherever the logic applies.

### CODE REVIEW

File: ecs/client.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	RegisterTaskDefinition(context.Context, *ecs.RegisterTaskDefinitionInput) (*ecs.RegisterTaskDefinitionOutput, error)
+	// DeregisterTaskDefinition deregisters an existing ECS task definition.
+	DeregisterTaskDefinition(context.Context, *ecs.DeregisterTaskDefinitionInput) (*ecs.DeregisterContainerInstanceOutput, error)
+	// RunTask runs a registered task.
+	RunTask(context.Context, *ecs.RunTaskInput) (*ecs.RunTaskOutput, error)
+}
+
+// ECSClient provides an implementation that wraps the ECS API.
+type ECSClient struct {
+}
```
Comment (julianedwards) [inline ecs/client.go:27]: Nit: can we move this up a line?

**Review by julianedwards — COMMENTED** [review]
Two things: 
1. I am not sure that we should add this interface to pail unless we refactor the entire repo and update what pail actually is. Currently, it is an interface for blob storage containers.
2. To increase decoupling, could this instead be a generic container interface that could be implemented for any container orchestration service? Similar to how the `Pail` interface can be implemented for S3, GridFS, a local filesystem, etc.

File: ecs/client.go (hunk 1 above)
Comment (Kimchelly) [reply ecs/client.go]: Done.

**Review by Kimchelly — COMMENTED** [review]
1. I'm okay with moving it into a separate repo, but I don't really know how it should be organized. One service = one repo? One repo containing a bunch of random AWS services?
2. I think thihs would be difficult because the models of container management aren't all that similar (i.e. none of the parameters really line up), so it's not like pail's S3 interface where you can assume all the things behave kind of like a filesystem. I think even if made an abstraction layer that supported multiple different implementations, it would be hard to provide the same guarantees across them, which basically defeats the point of an abstraction. I've seen the way that the cloud providers "implement" the same interface and basically they're not very compatible (e.g. the "static" cloud provider isn't even a provider and supports a tiny subset of the operations). The problem is that we assume that we're using one specific AWS service and that assumption gets baked into the interface more and more heavily into the interface as time goes on (making it a useless abstraction).
Basically, if we were to ever implement an alternative container orchestration service based on something other than ECS (namely Kubernetes), I think we would have to re-write everything.

**Comment by Kimchelly** [comment]
Closing this since I'm going to move it into a separate repo ([cocoa](https://github.com/evergreen-ci/cocoa)).
